feat: 채널 변경 상태에 따른 모달 및 상태 정보 변경 기능 구현

devjaeyoon opened the pull request:
### ✨ 이슈 번호

- #116

### 📌 설명

현재 채널이 광고 재생으로 인해 이동하기 전 채널인지, 광고가 재생되어 이동한 채널인지에 따라 설정, 모달 정보 변경 기능을 구현한 Pull Request입니다.

### 📃 작업 사항

- [x] 현재 채널이 광고 재생으로 인해 이동하기 전 채널인지, 광고가 재생되어 이동한 채널인지에 따라 설정, 모달 정보 변경 기능을 구현합니다.

### 💭 리뷰 사항 반영

- [x] `zustand` 선택적 구독 추가

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

devjaeyoon committed:
chore: return 앞 개행 추가

devjaeyoon committed:
feat: 광고 감지 시 채널 변경 상태 플래그 처리 로직 추가

광고 탐지로 채널이 자동 전환되면 setIsChannelChanged를 true로 설정하고,
광고 종료 후 원래 채널로 복귀 시 false로 초기화하도록 처리함.
MiniPlayer 및 재생 상태 유지 UX 흐름을 위한 상태 관리 개선 목적.

devjaeyoon committed:
feat: 채널 변경 여부에 따라 광고 제보 옵션 필터링 기능 추가

ReportHeader에서 isChannelChanged 전역 상태를 AdReportModal에 전달함
isChannelChanged가 true일 경우에만 NOT_AD 옵션이 노출되도록 필터링 적용

faber-h commented on src/components/header/ReportHeader.jsx:
```diff
@@ -7,7 +7,7 @@ import AdReportModal from "@/components/AdReportModal";
 
 const ReportHeader = () => {
-  const channelId = useChannelStore((state) => state.selectedChannelId);
+  const { isChannelChanged, selectedChannelId } = useChannelStore();
```
```suggestion
  const isChannelChanged = useChannelStore((state) => state.isChannelChanged);
  const selectedChannelId = useChannelStore((state) => state.selectedChannelId);
```
useChannelStore()를 직접 호출해 구조분해할 경우 store의 모든 상태를 구독하게 되어
다른 값이 바뀌어도 ReportHeader 컴포넌트가 불필요하게 리렌더링될 수 있습니다.

리렌더링을 최소화하기 위해 선택적 구독(selector)을 사용하는 것이 어떨까요?
두 분의 의견이 궁금합니다!
@Eun0713 @devjaeyoon

devjaeyoon replied:
좋은 의견 감사드립니다. `useChannelStore()`를 사용하면 `useChannelStore`의 `interestChannelIds`가 변경될 때 불필요하게 리렌더링되는 상황이 일어날 수 있겠네요. 해당 사항 반영하겠습니다!

d0-neii replied:
저도 `selector` 방식이 더 좋다고 생각해요!
전체 store를 구조분해하면 다른 값 변경 시에도 리렌더링될 수 있어서 필요한 값만 구독하는 게 더 효율적인 것 같습니다😊

devjaeyoon committed:
refactor: ReportHeader에 zustand selector 적용하여 리렌더링 최소화

useChannelStore 구조 분해 대신 selector 방식으로 상태 구독하도록 수정함
interestChannelIds 등 불필요한 상태 변경 시 리렌더링 방지 목적

faber-h reviewed (approved):
LGTM 🔥

Eun0713 reviewed (approved):
LGTM✨
